src/esphome/light/light_transformer.cpp

```diff
@@ -59,7 +59,7 @@ LightColorValues LightFlashTransformer::get_end_values() { return this->get_star
 LightFlashTransformer::LightFlashTransformer(uint32_t start_time, uint32_t length, const LightColorValues &start_values,
                                              const LightColorValues &target_values)
     : LightTransformer(start_time, length, start_values, target_values) {}
-bool LightFlashTransformer::is_continuous() { return false; }
+bool LightFlashTransformer::is_continuous() { return true; }
```

Comment:
Flashes are not continuous transformers 

> is_continuous(): Whether the output needs to be written in every loop cycle.

If this is to fix some behavior with flash it should be done in the state.

Reply:
From what I can see in the code, if is_continuous is false, the flash is applied in the first iteration of loop, but then the loop ends and it's never reverted back to the previous state. What is your suggestion of how this should be? (is_continuous is very confusing as a property, since you could argue that transitioning from brightness 50% to brightness 50% over 10 seconds is continuous, but then so is a flash, since it has a duration...).

Reply:
> but then the loop ends and it's never reverted back to the previous state

See line 63, at the end of a transformer the end values are applied

```cpp
this->remote_values_ = this->values_ = this->transformer_->get_end_values();
```

> but then so is a flash, since it has a duration...

Continuous here means that the transformer has a new value in each loop cycle, like it is for a transition. Flashes only need to override the state two times: At the beginning of the flash, and at the end of the flash.

Reply:
The flash transformer never says is_finished, because it's not updated ever again, since it's not continuous :-)

Reply:
Ah ok now I understand, yes that makes sense :)

On second thought, I think we can just remove `is_continuous` - while a transformer is active it should have complete control of the light. So remove `is_continuous` and change line 204 to 
```cpp
if (this->next_write_ || this->transformer_ != nullptr) {
```